flutter-idea/src/io/flutter/run/daemon/DevToolsServerTask.java

```diff
@@ -0,0 +1,399 @@
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+class DevToolsServerTask extends Task.Backgroundable {
+  private static final Logger LOG = Logger.getInstance(DevToolsServerTask.class);
+  public static final String LOCAL_DEVTOOLS_DIR = "flutter.local.devtools.dir";
+  public static final String LOCAL_DEVTOOLS_ARGS = "flutter.local.devtools.args";
+  private @NotNull Project project;
+  private final AtomicReference<CompletableFuture<DevToolsInstance>> devToolsFutureRef;
+
+  private DaemonApi daemonApi;
+  private ProcessHandler process;
+
+  public DevToolsServerTask(@NotNull Project project,
+                            @NotNull String title,
+                            AtomicReference<CompletableFuture<DevToolsInstance>> devToolsFutureRef) {
+    super(project, title, true);
+    this.project = project;
+    this.devToolsFutureRef = devToolsFutureRef;
+  }
+
+  @Override
+  public void run(@NotNull ProgressIndicator progressIndicator) {
+    try {
+      progressIndicator.setFraction(30);
+      progressIndicator.setText2("Init");
+
+      // If DevTools is not supported, start the daemon instead.
+      final boolean dartDevToolsSupported = dartSdkSupportsDartDevTools();
+      if (!dartDevToolsSupported) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Daemon set-up");
+        setUpWithDaemon();
+        return;
+      }
+
+      // If we are in a Bazel workspace, start the server.
+      // Note: This is only for internal usages.
+      final WorkspaceCache workspaceCache = WorkspaceCache.getInstance(project);
+      if (workspaceCache.isBazel()) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Starting server");
+        setUpWithDart(createCommand(workspaceCache.get().getRoot().getPath(), workspaceCache.get().getDevToolsScript(),
+                                    ImmutableList.of("--machine")));
+        return;
+      }
+
+      // This is only for development to check integration with a locally run DevTools server.
+      // To enable, follow the instructions in:
+      // https://github.com/flutter/flutter-intellij/blob/master/CONTRIBUTING.md#developing-with-local-devtools
+      final String localDevToolsDir = Registry.stringValue(LOCAL_DEVTOOLS_DIR);
+      if (!localDevToolsDir.isEmpty()) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Starting local server");
+        setUpLocalServer(localDevToolsDir);
+      }
+
+      // If the Dart plugin does not start DevTools, then call `dart devtools` to start the server.
+      final Boolean dartPluginStartsDevTools = true;
```

Comment:
Is there a Dart plugin version we want to gate this on? Or was this from long ago enough that it is safe to remove? 

Here's when the Dart Plugin began setting the host and port: https://github.com/JetBrains/intellij-plugins/commit/91822865c5a97915d68f0345486b53fe8cf4d17c

Reply:
I'll defer to @jwren but I don't think we're compatible with any plugins that go that far back so I think not?

Reply:
No, don't gate it on the Dart Plugin version.